Service/DataService.php

```diff
@@ -2010,4 +2023,101 @@ public function createAndMapIndex(Environment $environment): void
             'name' => $environment->getAlias()
         ]);
     }
+
+    public function getDataLinks(string $contentTypesCommaList, array $businessIds): array
+    {
+        $items = [];
+        $ouuids = [];
+        foreach ($businessIds as $businessId) {
+            if (isset($this->cacheOuuids[$contentTypesCommaList][$businessId])) {
```

Comment:
use a cache object that handles the Ouuids. That way you can add new ons dynamically. And no need for the "else" branch!